usr/share/rear/lib/layout-functions.sh

```diff
@@ -548,11 +550,20 @@ get_device_mapping() {
 }
 
 # Get the size in bytes of a disk/partition.
+# For disks, use "sda" as argument.
 # For partitions, use "sda/sda1" as argument.
 get_disk_size() {
     local disk_name=$1
 
-    local block_size=$(get_block_size ${disk_name%/*})
+    if has_binary blockdev; then
+        # ${disk_name##*/} translates 'sda/sda1' into 'sda1' and 'sda' into 'sda'
+        blockdev --getsize64 /dev/${disk_name##*/} && return
+    fi
+
+    # Linux always considers sectors to be 512 bytes long. See the note in the
+    # kernel source, specifically, include/linux/types.h regarding the sector_t
+    # type for details.
+    local block_size=512
 
     retry_command test -r /sys/block/$disk_name/size || Error "Could not determine size of disk $disk_name"
```

Comment:
I think also in the get_disk_size the $1 disk_name parameter
must be always the basename (as in the get_block_size function)
because on my SLES12 test system the directories in /sys/block are
<pre>
# ls /sys/block
md0  md1  md2  sda  sdb  sr0  sr1
</pre>
i.e. only the basename is used as in /sys/block/sda/size
but there is nothing like /sys/block/dev/sda/size

Reply:
it isn't. When a partition is specified (e.g. `sda/sda1`), then it has to read `/sys/block/sda/sda1/size`.
get_block_size() function is different because it is non-sense asking for block size of a partition, but for hardening it's stripping everything in front of the blockdev (e.g. `/dev/sda`)

Reply:
@rmetrich 
thank you for the explanation!